<?xml version="1.0"?>
<launch>

    <group ns="lidar">
        <group ns="center">
            <node type="ground_removal.py" name="ground_removal" pkg="practice_7" output="screen" required="true">   
                <remap from="points_raw" to="/lidar_center/points_raw" />
            </node>
        </group>

        <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" required="true" />
        <node pkg="nodelet" type="nodelet" name="voxel_grid_filter" args="load pcl/VoxelGrid pcl_manager" output="screen" required="true">
            <remap from="~input" to="/detection/lidar/center/points_no_ground" />
            <remap from="~output" to="/detection/lidar/points_filtered" />
        </node>

        <node type="points_clusterer.py" name="points_clusterer" pkg="practice_7" output="screen" required="true" />
        <node type="cluster_detector.py" name="cluster_detector" pkg="practice_7" output="screen" required="true" />

    </group>

    <node type="ema_tracker.py" name="ema_tracker" pkg="practice_7" output="screen" required="true" >
        <remap from="detected_objects" to="/detection/lidar/detected_objects" />
        <remap from="tracked_objects" to="final_objects" />
    </node>

    <node type="detected_objects_visualizer.py" name="final_objects_visualizer" pkg="practice_7" output="screen" required="true">
        <remap from="detected_objects" to="final_objects" />
        <remap from="detected_objects_markers" to="final_objects_markers" />
    </node>

    <rosparam command="load" file="$(find practice_7)/config/detection.yaml"/>

</launch>